src/VisualStudio/LiveShare/Impl/GotoDefinitionWithFarHandler.Exports.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using System;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+{
+    [ExportLspRequestHandler(LiveShareConstants.RoslynContractName, Methods.TextDocumentDefinitionName)]
+    [Obsolete("Used for backwards compatibility with old liveshare clients.")]
```

Comment:
i'm a little confused.  what old liveshare client is there?

Reply:
The liveshare client unfortunately used to have a hardcoded dependency on the server it was connecting to exporting handlers with the `RoslynContractName` for all 3 languages (c#/vb/ts).  We removed that hardcoded dependency, but liveshare clients on previous VS versions (e.g. everything before 16.3) still need to be able to connect to a 16.3 server.  So we have to export the handlers with that specific contract name for all three languages.